extra-files/split_logs.py

```diff
@@ -0,0 +1,90 @@
+# -*- encoding: utf-8 -*-
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+from xml.dom import minidom
+import datetime
+import argparse
+import os
+import re
+
+def get_logs(path):
+    logs = {} # filename -> lines[]
+    for file_name in os.listdir(path):
+        if not log_name.startswith("guest-test-"):
+            continue
+        with open(path + file_name, 'r') as f:
+            logs[file_name] = f.readlines()
+    return logs
+
+def filter_logs_by_time(logs, start_time, end_time):
+    filtered_logs = {} # filename -> lines[]
+    timestamp_re = r'[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}'
+
+    def line_in_scope(line):
+        timestamp_match = re.search(timestamp_re, line)
+        if timestamp_match is None:
+            return False
+        else:
+            timestamp = datetime.datetime.fromisoformat(
+                timestamp_match.group(0))
+            return start_time <= timestamp <= end_time
+
+    for log in logs:
+        lines = logs[log]
+        lines_to_keep = list(filter(line_in_scope, lines))
+        if len(lines_to_keep) > 0:
+            filtered_logs[log] = lines_to_keep
+
+    return filtered_logs
+
+def main():
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('--junit-xml', required=True,
+                        help='Junit XML file containing test data.')
+    parser.add_argument('--xen-logs',
+                        default='/var/log/xen/console',
+                        help='Where /var/log/xen/console is located\
+                            (useful for testing in dev vm).')
+    parser.add_argument('--outdir', required=True,
+                        help='Where to output the split logs.')
+
+    args = parser.parse_args()
+    xml = minidom.parse(args.junit_xml)
```

Comment:
This would be too easy... The junit XML (at least as generated by nose2) is in fact multiple concatenated documents (multiple root elements).

https://openqa.qubes-os.org/tests/27001#step/system_tests/39

Reply:
Malformed XML :'( I was testing with a local XML I had generated which had only one test suite. I've now tested with [a real one](https://openqa.qubes-os.org/tests/26999).

Rebased and fixed this. But it's still not ready for review. I found another issue.